exercises/sieve/src/example.c

```diff
@@ -1,50 +1,41 @@
-#include <string.h>
-#include <stdlib.h>
-#include <math.h>
 #include "sieve.h"
 
-unsigned int sieve(const unsigned int limit, primes_array_t primes)
-{
-   unsigned int number_of_primes = 0;
-
-   // clear the results
-   memset(primes, 0, sizeof(*primes));
+#include <stdbool.h>
+#include <stdlib.h>
 
-   if (limit > 1) {
-      //allocate 1 more than limit for convenience so the number and the index are same
-      unsigned char *number_is_prime = malloc(limit + 1);
-      memset(number_is_prime, 1, limit + 1);
+uint32_t sieve(uint32_t limit, uint32_t * primes, size_t max_primes)
+{
+   if (limit <= 1) {
+      // no primes that small
+      return 0;
+   }
+   // allocate one boolean for each number in [0,limit]
+   bool *data = calloc(limit + 1, sizeof(*data));
```

Comment:
Can you use a better name than the rather generic `data`?

Reply:
Changed e7c5ba9a11afd4640fc1f573d047aa56c426e8f1